packages/core/src/graph/graph.ts

```diff
@@ -1783,14 +1785,52 @@ export default abstract class AbstractGraph extends EventEmitter implements IAbs
       comboConfig = combo;
     }
 
-    // step2: 更新 children，根据类型添加 comboId 或 parentId
+    // step 2: Pull children out of their parents
+    let comboTrees = this.get('comboTrees');
+    const isChildById = new Set(children);
+    const pulledComboTreesById = new Map();
```

Comment:
Well, since I was already using `Set` before, I decided to just go for it. I could use an object for both I guess...

Reply:
Hmm... no, I'll keep using `Set` and `Map`. I'd have to change my code to:
```
    const childrenIdsSet = Object.create(null) as Record<string, boolean>;
    const pulledComboTreesById = Object.create(null) as Record<string, ComboTree>;
```

which is not ideal. I need to use `Object.create(null)` because I don't want keys inherited from `Object`'s prototype to be mistaken for valid keys (i.e. `pulledComboTreesById[id]` must not be unduly truthy)

Also, I'd have to initialize `childrenIdsSet` with `true` values for each element in the dictionary, which is a pain since we don't have `_.keyBy`...

No, let's keep using `Set` and `Map`. They are allowed, as shown in https://github.com/sirensolutions/G6/blob/master/packages/pc/src/util/combo-util.ts#L45-L46

Reply:
> I need to use Object.create(null) because I don't want keys inherited from Object's prototype to be mistaken for valid keys

I'm included to think you're more likely to get a false truthy from Map's prototype, but I don't mind either way.
![image](https://user-images.githubusercontent.com/7104356/195869731-eef57658-119c-4b23-8afb-b41945703cfb.png)